Tests/Tests/STPColorUtilsTest.m

```diff
@@ -103,38 +103,42 @@ - (void)testAllColorSpaces {
         }
     };
 
-    CFStringRef colorSpaceNames[] = {
-      kCGColorSpaceSRGB,
-      kCGColorSpaceDCIP3,
-      kCGColorSpaceROMMRGB,
-      kCGColorSpaceITUR_709,
-      kCGColorSpaceDisplayP3,
-      kCGColorSpaceITUR_2020,
-      kCGColorSpaceGenericRGB,
-      kCGColorSpaceGenericXYZ,
-      kCGColorSpaceLinearGray,
-      kCGColorSpaceLinearSRGB,
-      kCGColorSpaceGenericCMYK,
-      kCGColorSpaceGenericGray,
-      kCGColorSpaceACESCGLinear,
-      kCGColorSpaceAdobeRGB1998,
-      kCGColorSpaceExtendedGray,
-      kCGColorSpaceExtendedSRGB,
-      kCGColorSpaceGenericRGBLinear,
-      kCGColorSpaceExtendedLinearGray,
-      kCGColorSpaceExtendedLinearSRGB,
-      kCGColorSpaceGenericGrayGamma2_2,
-    };
+    NSArray *colorSpaceNames = @[
+      (__bridge NSString *)kCGColorSpaceSRGB,
+      (__bridge NSString *)kCGColorSpaceDCIP3,
+      (__bridge NSString *)kCGColorSpaceROMMRGB,
+      (__bridge NSString *)kCGColorSpaceITUR_709,
+      (__bridge NSString *)kCGColorSpaceDisplayP3,
+      (__bridge NSString *)kCGColorSpaceITUR_2020,
+      (__bridge NSString *)kCGColorSpaceGenericRGB,
+      (__bridge NSString *)kCGColorSpaceGenericXYZ,
+      (__bridge NSString *)kCGColorSpaceLinearSRGB,
+      (__bridge NSString *)kCGColorSpaceGenericCMYK,
+      (__bridge NSString *)kCGColorSpaceGenericGray,
+      (__bridge NSString *)kCGColorSpaceACESCGLinear,
+      (__bridge NSString *)kCGColorSpaceAdobeRGB1998,
+      (__bridge NSString *)kCGColorSpaceExtendedGray,
+      (__bridge NSString *)kCGColorSpaceExtendedSRGB,
+      (__bridge NSString *)kCGColorSpaceGenericRGBLinear,
+      (__bridge NSString *)kCGColorSpaceExtendedLinearSRGB,
+      (__bridge NSString *)kCGColorSpaceGenericGrayGamma2_2,
+      ];
+
+    if (@available(iOS 10.0, *)) {
+        colorSpaceNames = [colorSpaceNames arrayByAddingObjectsFromArray:@[
+                                                                           (__bridge NSString *)kCGColorSpaceLinearGray,
+                                                                           (__bridge NSString *)kCGColorSpaceExtendedLinearGray,
+                                                                           ]];
+    }
 
-    int colorSpaceCount = sizeof(colorSpaceNames) / sizeof(colorSpaceNames[0]);
-    for (int i = 0; i < colorSpaceCount; ++i) {
+    for (NSString *name in colorSpaceNames) {
         // CMYK is the only one where all 1's results in a dark color
-        testColorSpace(colorSpaceNames[i], colorSpaceNames[i] != kCGColorSpaceGenericCMYK);
+        testColorSpace(name, name != (__bridge NSString *)kCGColorSpaceGenericCMYK);
```

Comment:
Interesting, I would have assumed we needed to use `[NSString isEqualToString:]` but the tests are passing anyway?

Reply:
I'm less confident about keeping this as a pointer equality check with the bridging involved. I think it's probable that `CFStringRef` is always toll-free bridged to `NSString` and this'll be fine, but I think it's also plausible that some future runtime changes that.

Reply:
Yeah I agree it's not smart to leave it as a pointer check. I'm more wondering why it still works. Maybe this constant is always mapped to the same memory address. Anyway, I'll switch it to the usual `isEqualToString`.